pkg/controllers/management/aks/aks_cluster_handler_test.go

```diff
@@ -70,8 +77,8 @@ func Test_onClusterChange_Default(t *testing.T) {
 	if err != nil {
 		t.Errorf("error running onClusterChange: %s", err)
 	}
-	if cluster.Status.Conditions[0].Status != "Unknown" {
-		t.Errorf("status should be Unknown and cluster returned successfully")
+	if cluster.Status.Conditions[0].Status != mockConditionUnknown.Status {
```

Comment:
Sorry, I should've clarified, the issue is the lefthand side of the comparison. I don't even know what condition that's supposed to be without looking at the yaml. Something like the following would be alot more idiomatic: 
```
	if !rke2.Provisioned.IsTrue(cluster) || !rke2.Updated.IsUnknown(cluster) {
		t.Errorf("provisioned status should be True, updated status should be Unknown and cluster returned successfully: %s", err)
	}
```